File: src/web/magmac/api/head/RangeHead.ts
import { None } from "../../../magmac/api/None";
import { Option } from "../../../magmac/api/Option";
import { Some } from "../../../magmac/api/Some";
export class RangeHead {
	RangeHead(length : int) : public {this.length=length;this.counter=0;;}
	next() : Option<Integer> {if(true){ value : int=this.counter;this.counter++;return new Some<>( value);;}if(true){ return new None<>( );;};}
}

File: packages/compiler-java/src/main/java/magmac/api/head/RangeHead.java
package magmac.api.head;

import magmac.api.None;
import magmac.api.Option;
import magmac.api.Some;

/**
 * {@link Head} producing a range of integers.
 */

public class RangeHead implements Head<Integer> {
    private final int length;
    private int counter;

    public RangeHead(int length) {
        this.length = length;
        this.counter = 0;
    }

    @Override
    public Option<Integer> next() {
        if (this.counter < this.length) {
            var value = this.counter;
            this.counter++;
            return new Some<>(value);
        }
        else {
            return new None<>();
        }
    }
}
